spec/semantics.md

```diff
@@ -0,0 +1,33 @@
+## Phy Specification
+
+This document describes the semantics of the source language.
+
+A *program* consists of a single *module*. A *module* consists of zero or more
+*declarations*.
+
+A *declaration* introduces a new entity, one that can later be referred to with
+the given name.
+
+An *expression* is a term or a control-flow instruction. All expressions have a
+*type*.
+
+A *statement* is a computation without a type.
```

Comment:
I guess this would cover definitions like procedures, which we're not consider to be `unit`.